### PR TITLE
Binary Node Completion

### PR DESCRIPTION
**Pull request by MEEPofFaith**
Hopefully no bugs suddenly show up and aaaaaAAAAAAAAAAAAAAAAAA me.

https://user-images.githubusercontent.com/54301439/133943833-bffd4eda-d769-457e-8f87-9391ed405ce8.mp4

https://user-images.githubusercontent.com/54301439/133943840-4669b304-e89c-47c2-a76f-82cd02780bc1.mp4

### CODE REVIEW

File: src/esoterum/world/blocks/binary/BinaryNode.java [hunk 1]
```diff
@@ -1,56 +1,190 @@
...
         range = linkRange;
+        configurable = true;
 
         outputs = new boolean[]{true, true, true, true};
         inputs = new boolean[]{true, true, true, true};
 
+        //point2 config is relative
+        config(Point2.class, (BinaryNodeBuild tile, Point2 i) -> tile.link = Point2.pack(i.x + tile.tileX(), i.y + tile.tileY()));
+        //integer is not
+        config(Integer.class, (BinaryNodeBuild tile, Integer i) -> tile.link = i);
```
Comment (MEEPofFaith) [inline src/esoterum/world/blocks/binary/BinaryNode.java:33]: Literally copied from [ItemBridge.java](https://github.com/Anuken/Mindustry/blob/master/core/src/mindustry/world/blocks/distribution/ItemBridge.java).

**Comment by MEEPofFaith** [comment]
https://user-images.githubusercontent.com/54301439/133944775-b0ef5eff-306f-4e89-8162-5213057d4a6d.mp4

**Comment by MEEPofFaith** [comment]
Up next: Fixing breaking.

**Comment by MEEPofFaith** [comment]
Nodes being destroyed, picked up, deconstructed, or somehow removed some other way will disconnect the node it's connected to.

**Comment by MEEPofFaith** [comment]
aaa more buggy behavior found

**Comment by MEEPofFaith** [comment]
https://user-images.githubusercontent.com/54301439/133945180-b90aa06a-bc23-4689-8c76-426870dcbd73.mp4

**Comment by MEEPofFaith** [comment]
The only way I can think of to fix this is to make node connections not save in schematics, but I don't want to do that.

**Comment by MEEPofFaith** [comment]
https://user-images.githubusercontent.com/54301439/133945376-bfe7c285-fc09-4bb2-b04c-e6af3614e777.mp4

**Comment by Goobrr** [comment]
so are you done?

**Comment by MEEPofFaith** [comment]
now I am

**Comment by MEEPofFaith** [comment]
~~At least, until an unforseen bug shows up.~~

**Pull request merged**